lib/auth/cognito_stub_client.rb

```diff
@@ -1,44 +1,60 @@
 class CognitoStubClient
   # TODO Turn stub_user_hash into a JWT token which
   # can be returned by the stub client
-  def self.stub_user_hash(role:, email_domain: "test.com")
-    { username: '00000000-0000-0000-0000-000000000000', user_attributes:
-      [
-        { name: 'sub', value: '00000000-0000-0000-0000-000000000000' },
-        { name: 'custom:roles', value: role },
-        { name: 'email_verified', value: 'true' },
-        { name: 'phone_number_verified', value: 'true' },
-        { name: 'phone_number', value: '+447000000000' },
-        { name: 'given_name', value: 'Daenerys' },
-        { name: 'family_name', value: 'Targaryen' },
-        { name: 'email', value: "daenerys.targaryen@#{email_domain}" }
-      ],
-    preferred_mfa_setting: 'SOFTWARE_TOKEN_MFA',
-    user_mfa_setting_list: %w[SOFTWARE_TOKEN_MFA] }
+  def self.stub_user_hash(role:, email_domain: "test.com", groups: %w[devs])
+    {
+      'sub' => '00000000-0000-0000-0000-000000000000',
+      'custom:roles' => role,
+      'cognito:groups' => groups,
+      'email_verified' => true,
+      'phone_number_verified' => true,
+      'cognito:username' => '00000000-0000-0000-0000-000000000000',
+      'given_name' => 'Daenerys',
+      'cognito:roles' => ['arn:aws:iam::000000000000:role/daenerys.targaryen-read-only'],
+      'event_id' => '00000000-0000-0000-0000-000000000000',
+      'token_use' => 'id',
+      'auth_time' => Time.now,
+      'phone_number' => '+447000000000',
+      'exp' => 1.week.from_now,
+      'iat' => Time.now,
+      'family_name' => 'Targaryen',
+      'email' => "daenerys.targaryen@#{email_domain}",
```

Comment:
What happens when the breaker of chains joins GDS for realz?

Reply:
We quake with fear!